compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/SwitchGenerator.kt

```diff
@@ -348,7 +348,27 @@ class SwitchGenerator(private val expression: IrWhen, private val data: BlockInf
         override fun shouldOptimize() = cases.size > 1
 
         override fun genSwitch() {
-            subject.accept(codegen, data).materialize()
+            // Do not generate line numbers for the table switching. In particular,
+            // the subject is extracted from the condition of the first branch which
+            // will give the wrong stepping behavior for code such as:
+            //
+            // when {
+            //   x == 42 -> 1
+            //   x == 32 -> 2
+            //   x == 24 -> 3
```

Comment:
Could you also add some tests for multiline `case`, e.g.

            when {
               x == 42 -> 1
               x == 32 -> 
                 2
               x == 24 -> 3

Reply:
Sorry for the late reply on this. Vacation got in the way. :) 

Good idea. Done.